Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -150,9 +156,9 @@ private void OnCleanForensicsDoAfter(EntityUid uid, ForensicsComponent component
                 targetComp.Residues.Add(string.IsNullOrEmpty(residue.ResidueColor) ? Loc.GetString("forensic-residue", ("adjective", residue.ResidueAdjective)) : Loc.GetString("forensic-residue-colored", ("color", residue.ResidueColor), ("adjective", residue.ResidueAdjective)));
         }
 
-        public string GenerateFingerprint()
+        public string GenerateFingerprint(int length = 16)
         {
-            var fingerprint = new byte[16];
+            var fingerprint = new byte[length];
```

Comment:
```suggestion
            var fingerprint = new byte[Math.Clamp(length, 0, 255)];
```

Otherwise this will throw an error due to the mismatch in allowable inputs and what the function will actually accept, since this is a public function, and anyone anywhere can just throw an integer at it. Bytes can be any integer between 0 and 255, whereas a standard Integer can be as high as 18446744073709551615, and also includes negative numbers.